File: typescript-javascript/algorithms/easy/miniMaxSum.ts
/**
 * @author Rithie Natan Carvalhaes Prado
 * @date 2026-05-19
 * Last modified: 2026-06-11
 * @description Mini-Max Sum - HackerRank
 * @see {@link https://www.hackerrank.com/challenges/mini-max-sum/problem}
 */
//---------- functions ----------
/**
 * Calculates and prints the minimum and maximum sums of an array.
 * @param {number[]} arr - The array of numbers.
 */
function miniMaxSum(arr: number[]): void
{
    arr.sort((a, b) => a - b);
    
    const min = arr.slice(0, arr.length - 1).reduce((a, b) => a + b);
    const max = arr.slice(1, arr.length).reduce((a, b) => a + b);
    
    console.log(min + " " + max);
}//end miniMaxSum()

File: go/algorithms/easy/miniMaxSum.go
/*
Author: Rithie Natan Carvalhaes Prado
Date: 2026-06-13
Last modified: 2026-06-13
Description: Plus Minus - HackerRank
See: https://www.hackerrank.com/challenges/plus-minus/problem
*/
//--------- package ---------
package easy

//--------- Imports ---------
import (
	"fmt"
	"slices"
)

//--------- functions ---------
// miniMaxSum calculates the minimum and maximum sum and prints them.
func miniMaxSum(arr []int32) {
	slices.Sort(arr);
	
	var min []int32 = arr[0:len(arr)-1];
	var max []int32 = arr[1:];

	var sumMin, sumMax int64 = 0, 0;

	for i := 0; i < len(min); i++ {
		sumMin += int64(min[i]);
		sumMax += int64(max[i]);
	}

	fmt.Printf("%d %d\n", sumMin, sumMax);
}//miniMaxSum()